### ISSUE TITLE
Any way to hide Pydantic 2.0 class internals? 

### ISSUE THREAD

**Issue by aaronsteers** [mitmproxy/pdoc#715]
#### Problem Description

My class declaration has no members declared explicitly. However, the `pdoc` text generated includes some garbage internals from the Pydantic class, while not actually documenting any of the superclass's members. I would like to just globally suppress these:

- `model_config`
- `model_post_init`
- `model_fields`
- `model_computed_fields`

My class documentation index looks like this since migrating to Pydantic 2.0:

> ![image](https://github.com/user-attachments/assets/9a6d7fc7-47cf-4fc6-80dc-911a26d308b8)

Since I'm not declaring these directly in my own code, I don't have a way to suppress them with a `@private` annotation or similar.

#### Steps to reproduce the behavior:

Create a class from Pydantic 2.0 `BaseModel`.

#### System Information

Paste the output of "pdoc --version" here.

```
pdoc: 14.5.1
Python: 3.10.12
Platform: macOS-14.5-arm64-arm-64bit
```

**Comment by elliot-100** [comment]
This seems to work (with a caveat re #714 ) if you add it to the superclass:
```python
@property
def model_fields(self):
    """@private"""
    pass

**Comment by mhils** [comment]
My suggestion here would be to use a custom template that overrides `is_public`:

https://github.com/mitmproxy/pdoc/blob/8939754cbee82ad08b939bccfa7123d7bcb4fcc5/examples/custom-template/module.html.jinja2#L21-L32

**Comment by JPHutchins** [comment]
> My suggestion here would be to use a custom template that overrides `is_public`:
> 
> https://github.com/mitmproxy/pdoc/blob/8939754cbee82ad08b939bccfa7123d7bcb4fcc5/examples/custom-template/module.html.jinja2#L21-L32

I've found that this example syntax is not quite right.  The inline comments of default_is_public state that false is given by an empty string rather than a literal boolean.

```jinja2
{#
We can also adjust which members are documented by overriding the is_public macro.
#}
{% macro is_public(doc) %}
    {% if doc.name.startswith("model_") %}
        {# False #}
    {% else %}
        {{ default_is_public(doc) }}
    {% endif %}
{% endmacro %}
```

I'm not sure what's right, but this is what finally worked for me.

**Comment by JPHutchins** [comment]
I'll be looking into filtering at the BaseModel inheritance as well since it is irrelevant to users.

<img width="598" alt="image" src="https://github.com/user-attachments/assets/ed158872-a351-4bd1-b0ce-d5caaae99051">

**Comment by JPHutchins** [comment]
Thinking about it more, what I'm really looking for is filtering out inheritance that's not in the repo.  That is, it can't be linked to.  For example, it's not helpful to tell my users about all the builtin methods on a bytes object 🤣 

<img width="594" alt="image" src="https://github.com/user-attachments/assets/b98a5b53-c7e0-4c55-9ec3-a4f193560559">

**Comment by mhils** [comment]
> Thinking about it more, what I'm really looking for is filtering out inheritance that's not in the repo. 

This actually sounds like pretty sensible default behavior! Let's ship a release with this and see if someone thinks different. :)

**Comment by JPHutchins** [comment]
I have since tried mkdocs (mkdocstrings) and can say that it errs in the opposite direction, where docstrings and members are not inherited.  There's a plugin that CAN inherit docstrings, but only from within the repo, which caused me all sorts of problems.  Eventually I did some LLM script hack garbage: https://github.com/intercreate/smpclient/blob/d318220d9dd70f1861cae73d2b7b24a6c3388132/docs/_generate_requests_docstrings.py

I know it's useless to say, but this all needs to be standardized in the language; pdoc and mkdocstrings are doing a great job, but it's really hard without any specs!

It led me to find PEP 727, which I support: https://discuss.python.org/t/pep-727-documentation-metadata-in-typing/32566
